relay.go

```diff
@@ -54,29 +54,20 @@ type Relay struct {
 	relays map[peer.ID]struct{}
 	mx     sync.Mutex
 
+	isAllowedToHop Acceptor
+
 	// atomic counters
 	streamCount  int32
 	liveHopCount int32
 }
 
-// RelayOpts are options for configuring the relay transport.
-type RelayOpt int
+// RelayOpts are function for configuring the relay transport.
+type RelayOpt func(*Relay) error
 
-var (
-	// OptActive configures the relay transport to actively establish
-	// outbound connections on behalf of clients. You probably don't want to
-	// enable this unless you know what you're doing.
-	OptActive = RelayOpt(0)
-	// OptHop configures the relay transport to accept requests to relay
-	// traffic on behalf of third-parties. Unless OptActive is specified,
-	// this will only relay traffic between peers already connected to this
-	// node.
-	OptHop = RelayOpt(1)
-	// OptDiscovery configures this relay transport to discover new relays
-	// by probing every new peer. You almost _certainly_ don't want to
-	// enable this.
-	OptDiscovery = RelayOpt(2)
-)
+// Acceptor return true if this conn is allowed to hop on this peer.
+// Its a `network.Stream` and not `peer.ID` because we excpect user to maybe
+// filter by by subnet or by transport.
+type Acceptor func(network.Stream) bool
```

Comment:
We discussed this with @stebalien and we think we also need the target peer as well.
So the filter should be `func(network.Stream, peer.ID) bool`.

Reply:
I think so, I just didn't put because we can get it by `s.Conn().RemotePeer()` and some filter (like subnet didn't require it), including this in the filter allow to do `s.Conn().RemotePeer()` or not depending of the need (reducing memory footprint).